hvplot/converter.py

```diff
@@ -1411,32 +1410,44 @@ def _apply_layers(self, obj):
                     else:
                         feature_obj = feature_obj.opts(scale=scale)
                 if feature_obj.group in ["Land", "Ocean"]:
-                    obj = feature_obj * obj  # Underlay land/ocean
-                else:
-                    obj = obj * feature_obj  # overlay everything else
-
-        if self.tiles:
-            tile_source = 'EsriImagery' if self.tiles == 'ESRI' else self.tiles
-            warning = ("{} tiles not recognized, must be one of: {} or a tile object".format(tile_source, sorted(hv.element.tile_sources)))
-            if tile_source is True:
-                tiles = hv.element.tiles.OSM()
-            elif tile_source in hv.element.tile_sources.keys():
-                tiles = hv.element.tile_sources[tile_source]()
-            elif tile_source in hv.element.tile_sources.values():
-                tiles = tile_source()
-            elif isinstance(tile_source, hv.element.tiles.Tiles):
-                tiles = tile_source
-            elif self.geo:
-                from geoviews.element import WMTS
-                if isinstance(tile_source, WMTS):
-                    tiles = tile_source
+                    # Underlay land/ocean
+                    obj = feature_obj.opts(projection=self.proj_crs) * obj
                 else:
-                    param.main.param.warning(warning)
-            else:
-                param.main.param.warning(warning)
+                    # overlay everything else
+                    obj = obj * feature_obj.opts(projection=self.proj_crs)
+
+        if self.tiles and not self.geo:
+            tiles = self._get_tiles(
+                self.tiles,
+                hv.element.tile_sources,
+                hv.element.tiles.Tiles
+            )
+            obj = tiles * obj
+        elif self.tiles and self.geo:
+            import geoviews as gv
+            tiles = self._get_tiles(
+                self.tiles,
+                gv.tile_sources.tile_sources,
+                (gv.element.WMTS, hv.element.tiles.Tiles),
+            )
             obj = tiles * obj
         return obj
 
+    def _get_tiles(self, source, sources, types):
+        tile_source = 'EsriImagery' if self.tiles == 'ESRI' else self.tiles
+        warning = f"{tile_source} tiles not recognized, must be one of: {sorted(sources)} or a tile object"
```

Comment:
I don't think this should be called warning. Also I think this should be put directly in the else clause for clarity.